Gems/SceneProcessing/Code/Source/Generation/Components/MeshOptimizer/MeshOptimizerComponent.cpp

```diff
@@ -261,6 +264,45 @@ namespace AZ::SceneGenerationComponents
         ).empty();
     }
 
+    /* static AZ::Outcome<SceneData::GraphData::CustomPropertyData::PropertyMap&> FindOrCreateCustomPropertyData(
```

Comment:
I was trying to wrap this logic in a function in a clean way that lets you skip optimization if an optimized node already exists. But putting the PropertyMap& into an outcome was somehow ending up in a copy of the PropertyMap instead of a reference to the existing PropertyMap. In this version of the draft PR, this logic exists loosely in the OptimizeMeshes function, which works. I'll take another stab at moving this into a function.